### PR TITLE
Expose last-token candidates context method

### PR DESCRIPTION
**Pull request by brittlewis12**
* adds call to `llama_get_logits`, which accesses the result of the most recent `llama_decode` call
> Token logits obtained from the last call to `llama_decode()`
> The logits for which `llama_batch.logits[i] != 0` are stored contiguously in the order they have appeared in the batch.
> Rows: number of tokens for which `llama_batch.logits[i] != 0`
> Cols: `n_vocab`

https://github.com/ggerganov/llama.cpp/blob/c35e586ea57221844442c65a1172498c54971cb0/include/llama.h#L844-L849

### CODE REVIEW

**Review by MarcusDunn — CHANGES_REQUESTED** [review]
this looks good. I'd prefer if the names matched what's in llama.h with docs explaining that it gets the last logits. Otherwise, this looks good.

**Comment by brittlewis12** [comment]
superseded by https://github.com/utilityai/llama-cpp-rs/pull/509
